airflow/jobs.py

```diff
@@ -1790,6 +1790,54 @@ class BackfillJob(BaseJob):
         'polymorphic_identity': 'BackfillJob'
     }
 
+    class _DagRunTaskStatus(object):
```

Comment:
If we are putting this into an class, would it be something to add a integrity check? Ie there is a total number of tasks. The sum of the counters should always be equal to number. And yes we have seen leakage in the past.

Might be outside of this PR though (no logic change).